site/content/en/docs/Advanced/allocator-service.md

```diff
@@ -0,0 +1,115 @@
+  Agones provides mTLS based allocator service that is accessible from outside the cluster using a load balancer. The service is deployed and scales independent to Agones controller.
+---
+
+Agones supports two APIs to allocate a game server, one is exposed as a Kubernetes API Extention, [GameServerAllocations](https://github.com/googleforgames/agones/blob/master/pkg/apis/allocation/v1/gameserverallocation.go) and the other, that is the main topic of the discussion here, is exposed as a REST API service with mTLS authentication, called agones-allocator. The latter is on [v1alpha1 version](https://github.com/googleforgames/agones/blob/master/proto/allocation/v1alpha1) starting on agones v1.1.
+
+agones-allocator is exposing an external IP through a load balancer. The IP is the allocator server's REST endpoint. For the http request to succeed, a client certificate should be provided that is in the authorization list of allocator service. In the following, the steps to get the allocation request working with agones-allocator is explained:
+
+1. Find the external IP
+2. Set up server TLS certificates
+3. Add client cert to authorization list
+4. Make a rest call to allocate a game server
+
+## Find the external IP
+
+The service is hosted under the same namespace as Agones controller. To find the external IP of your allocator service, replace agones-system namespace with the the namespace to which Agones is deployed and execute the following command:
+
+```bash
+kubectl get service agones-allocator -n agones-system
+```
+
+The output of the command should be like:
+
+<pre>
+NAME                        TYPE           CLUSTER-IP      <b>EXTERNAL-IP</b>     PORT(S)            AGE
+agones-allocator            LoadBalancer   10.55.251.73    <b>34.82.195.204</b>   443:30250/TCP      7d22h
+</pre>
+
+Store the IP in a variable to use as the server endpoint in the next sections:
+
+```bash
+EXTERNAL_IP=`kubectl get services agones-allocator -n agones-system -o jsonpath='{.status.loadBalancer.ingress[0].ip}'`
+```
+
+## Server TLS certificate
+
+Replace the default server TLS certificate with a certificate with CN and subjectAltName. There are multiple approaches to generate a certificate, including using CA. The following provides an example of generating a self-signed certificate using openssl.
+
+```bash
+#!/bin/bash
+
+TLS_KEY_FILE=tls.key
+TLS_CERT_FILE=tls.crt
+
+cat /etc/ssl/openssl.cnf <(printf "\n[SAN]\nsubjectAltName=IP:${EXTERNAL_IP}") > openssl.cnf
+
+openssl req -nodes -new -newkey rsa:2048 \
+    -keyout ${TLS_KEY_FILE} \
+    -out tls.csr \
+    -subj "/CN=${EXTERNAL_IP}/O=${EXTERNAL_IP}" \
+    -reqexts SAN \
+    -config openssl.cnf
+
+openssl x509 -req -days 365 -in tls.csr \
+    -signkey ${TLS_KEY_FILE} \
+    -out ${TLS_CERT_FILE} \
+    -extensions SAN \
+    -extfile openssl.cnf
+
+kubectl create secret tls allocator-tls -n agones-system --key=${TLS_KEY_FILE} --cert=${TLS_CERT_FILE} --dry-run -o yaml | kubectl apply -f -
```

Comment:
Why do a dry-run and then pass the output to kubectl apply instead of just doing the create secret without a dry run? Does it generate a different result?